source/logic/bounties.js

```diff
@@ -16,6 +18,18 @@ function setDB(database) {
 	db = database;
 }
 
+/**
+ * @param {{slotNumber: number, posterId: string, guildId: string} | string} bountyInfo
+ * @returns {Bounty?} the bounty, if it exists
+ */
+async function findBounty(bountyInfo) {
+	if (typeof bountyInfo === 'string') {
+		return await db.models.Bounty.findByPk(bountyInfo,  { include: db.models.Bounty.Company });
+	} else {
+		return await db.models.Bounty.findOne({ where: { userId: bountyInfo.posterId, companyId: guildId, slotNumber: bountyInfo.slotNumber, state: "open" }, include: db.models.Bounty.Company });
+	}
+}
```

Comment:
```suggestion
/**
 * @param {{slotNumber: number, posterId: string, guildId: string} | string} bountyInfo
 */
function findBounty(bountyInfo) {
	if (typeof bountyInfo === 'string') {
		return db.models.Bounty.findByPk(bountyInfo,  { include: db.models.Bounty.Company });
	} else {
		return db.models.Bounty.findOne({ where: { userId: bountyInfo.posterId, companyId: guildId, slotNumber: bountyInfo.slotNumber, state: "open" }, include: db.models.Bounty.Company });
	}
}
```
In this function, I prefer to omit the `async-await` because the only thing being awaited is the final return, which practically doesn't change things. Further, it's usually better to allow VS Code detect the return type of the function, as seeing something out of place can help find bugs, and it doesn't have potential for outdated documentation desyncs.

Minimally, the return type is incorrect: should be `Promise<Bounty?>` instead of `Bounty?`.